Add repairing and unit repaired sounds to ra service depo

abcdefg30 opened the pull request:
In TD and D2K it currently only plays the "repairing" sound when entering a service depo
(because they don't have "unit repaired" notifications).
Should that stay that way, or should I use `StartRepairingNotification = null;` as default?

abcdefg30 committed:
Add repairing and unit repaired sounds to fix

penev92 commented:
I will have to check the Dune files for such a notification when I get home. But in any case, leave the "Repairing" notification ;)

penev92 commented:
The code looks good to me. I will test it ingame later.

penev92 commented on OpenRA.Mods.Common/Activities/Repair.cs:
```diff
@@ -32,7 +33,12 @@ public override Activity Tick(Actor self)
 
 			if (health.DamageState == DamageState.Undamaged)
+			{
+				var repairsUnits = host.Info.Traits.Get<RepairsUnitsInfo>();
```
You have the same thing at  https://github.com/abcdefg30/OpenRA/blob/repairnotifications/OpenRA.Mods.Common/Activities/Repair.cs#L45
Cache this in the constructor for best results.

abcdefg30 commented:
Updated with suggestions and added [this](https://github.com/OpenRA/OpenRA/pull/7298/files#diff-aac7efdcce31c89aaecaea0ca366cc28R40).
> (penev) so is ```if (health.DamageState == DamageState.Undamaged)``` where the repairing is finished?
(abcdefg30) i think so
(penev) could you please put a comment above the if (unrelated to the PR, but I think it would be nice)

Phrohdoh commented:
I think that truth statement makes plenty of sense without the comment.

penev92 commented:
Comments are nice. They make even the simple code easier to read.

Phrohdoh commented:
`1 + 2; // add 1 and 2`

pchote commented:
Or, as it often looks after 3 years of tweaks:
> (1 + 7) % 3; // add 1 and 2

penev92 commented:
Sorryy, @abcdefg30, I guess I should ask you to remove that comment now.  :/

abcdefg30 commented:
No problem. Updated.

Mailaender commented on OpenRA.Mods.Common/Activities/Repair.cs:
```diff
@@ -17,11 +17,17 @@ namespace OpenRA.Mods.Common.Activities
 	{
+		readonly RepairsUnitsInfo repairsUnits;
 		int remainingTicks;
 		Actor host;
```
Can you make this `readonly`, too?

abcdefg30 replied:
Done.

Mailaender commented:
:+1: Works as promised in ra and no regressions in d2k and cnc.

penev92 commented:
> I will have to check the Dune files for such a notification when I get home

Sadly, I found no such notification in the game files :( But I remember something like that. Maybe I should check Youtube.

penev92 commented:
OK, I just checked the original game and there definitely is a "Unit repaired" notification in Dune 2000. I'd hate to block this, but can we please hold on merging this until I (or if someone else is interested) find it in the D2k files so that this PR is complete when merged?

penev92 commented:
Found it!
Can you please add "UnitRepaired: GANEW" to D2k's notifications.yaml and use it on the repair pad in structures.yaml?

abcdefg30 commented:
Updated.

obrakmann commented:
You'll probably want make the `SYRD` and `SPEN` emit that notification, too.

The barracks (`^Barracks` in `defaults.yaml`) in d2k could also use it.

abcdefg30 commented:
Updated.

obrakmann commented:
:+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/65054631a5668eed2035aee39944f133bca34ae0%5E...65054631a5668eed2035aee39944f133bca34ae0)
